identity/client/src/main/java/com/azure/identity/IdentityClient.java

```diff
@@ -204,21 +209,11 @@ private static AuthenticationContext createAuthenticationContext(ExecutorService
             payload.append("resource");
             payload.append("=");
             payload.append(URLEncoder.encode(resource, "UTF-8"));
-            if (objectId != null) {
-                payload.append("&");
-                payload.append("object_id");
-                payload.append("=");
-                payload.append(URLEncoder.encode(objectId, "UTF-8"));
-            } else if (clientId != null) {
+            if (clientId != null) {
                 payload.append("&");
                 payload.append("client_id");
                 payload.append("=");
                 payload.append(URLEncoder.encode(clientId, "UTF-8"));
```

Comment:
Same issue here in code that isn't changed - excessive string builder operations that could be made simpler

Reply:
made simpler